store/src/java/com/zimbra/cs/service/mail/CopyToFiles.java

```diff
@@ -55,120 +54,183 @@ public CopyToFiles(
+    // execute Files api call and flatten yield result
+    // maps every non-ServiceException to internal error
+    String nodeId =
+        Optional.ofNullable(
+                For(
+                        authCookieTry,
+                        destFolderIdTry,
+                        fileNameTry,
+                        contentTypeTry,
+                        attachmentStreamTry,
+                        attachmentSizeTry)
+                    .yield(
+                        (authCookie,
+                            destFolderId,
+                            fileName,
+                            contentType,
                             attachmentStream,
-                            attachmentSize).get())
-                .onFailure(ex -> mLog.debug(ex.getMessage()))
-                .mapFailure(Case($(ex -> !(ex instanceof ServiceException)),
-                    ex -> ServiceException.FAILURE("internal error.", ex)))
-                .get())
-        .orElseThrow(() -> ServiceException.FAILURE("got null response from Files server."))
-        .getNodeId();
+                            attachmentSize) ->
+                            filesClient.uploadFile(
+                                authCookie,
+                                destFolderId,
+                                fileName,
+                                contentType,
+                                attachmentStream,
+                                attachmentSize))
+                    .flatMap(x -> x)
+                    .onFailure(ex -> mLog.debug(ex.getMessage()))
+                    .mapFailure(
+                        Case(
+                            $(ex -> !(ex instanceof ServiceException)),
+                            ex -> ServiceException.FAILURE("internal error.", ex)))
+                    .get())
+            .orElseThrow(() -> ServiceException.FAILURE("got null response from Files server."))
+            .getNodeId();
 
     CopyToFilesResponse copyToFilesResponse = new CopyToFilesResponse();
     copyToFilesResponse.setNodeId(nodeId);
     return zsc.jaxbToElement(copyToFilesResponse);
   }
 
   /**
-   * Transforms the SOAP request in object representation
+   * Transforms the SOAP request in {@link CopyToFilesRequest} instance
    *
    * @param request soap {@link Element} as received from client
    * @return try of {@link CopyToFilesRequest}
    */
   private Try<CopyToFilesRequest> getRequestObject(Element request) {
     return Try.<CopyToFilesRequest>of(() -> JaxbUtil.elementToJaxb(request))
         .onFailure(ex -> mLog.debug(ex.getMessage()))
-        .mapFailure(Case($(instanceOf(Exception.class)),
-            ex -> ServiceException.PARSE_ERROR("Malformed request.", ex)));
+        .mapFailure(
+            Case(
+                $(instanceOf(Exception.class)),
+                ex -> ServiceException.PARSE_ERROR("Malformed request.", ex)));
```

Comment:
See suggestion above:
```suggestion
        .onFailure(Exception.class, ex -> ServiceException.PARSE_ERROR("Malformed request.", ex));
```

Reply:
@Polpetta I'm not sure this will introduce the behavior I was thinking about.
The idea is:

- to log the underlying exception
- map the exception to a specific ServiceException.

By using the onFailure(), which accepts a consumer, it will execute the consumer in case it matches but it will return the identity (this). So it will not map the exception to a ServiceException and the old failure will be preserved. 
Also some tests have been made to track this behavior and if I change the code as suggested they fail.(see shouldThrowServiceExceptionIfFileNotFound)

What I can do (but it's a style thing at this point) is:
```suggestion
        .mapFailure(
            Case(
                $(instanceOf(Exception.class)),
                ex -> {
                              mLog.debug(ex.getMessage());
                              return ServiceException.NOT_FOUND("File not found.", ex);
                            }));
```
and remove the onFailure entirely.

Reply:
Ooooh ok, I got the point. Nope, keep it like it is and resolve all the other corrections pleasee :+1: Resolve this one once you've read the comment